src/pages/FarmPage/FarmsList.tsx

```diff
@@ -267,67 +261,42 @@ const FarmsList: React.FC<FarmsListProps> = ({ bulkPairs, farmIndex }) => {
     : null;
 
-  useEffect(() => {
-    setStakingInfos(undefined);
-    setStakingDualInfos(undefined);
-    setTimeout(() => setPageLoading(false), 500); //load farms 0.5s after loading page
-  }, []);
-
   useEffect(() => {
     setPageIndex(0);
-    if (farmIndex === GlobalConst.farmIndex.LPFARM_INDEX) {
-      setStakingInfos(sortedLPStakingInfos.slice(0, LOADFARM_COUNT));
-    } else {
-      setStakingDualInfos(sortedStakingDualInfos.slice(0, LOADFARM_COUNT));
-    }
-    return () => {
-      setStakingInfos(undefined);
-      setStakingDualInfos(undefined);
-    };
-    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [stakingRewardAddress]);
 
-  useEffect(() => {
-    if (farmIndex === GlobalConst.farmIndex.LPFARM_INDEX) {
-      const currentStakingInfos = stakingInfos || [];
-      const stakingInfosToAdd = sortedLPStakingInfos.slice(
-        currentStakingInfos.length,
-        currentStakingInfos.length + LOADFARM_COUNT,
-      );
-      setStakingInfos(currentStakingInfos.concat(stakingInfosToAdd));
-    } else if (farmIndex === GlobalConst.farmIndex.DUALFARM_INDEX) {
-      const currentDualStakingInfos = stakingDualInfos || [];
-      const stakingDualInfosToAdd = sortedStakingDualInfos.slice(
-        currentDualStakingInfos.length,
-        currentDualStakingInfos.length + LOADFARM_COUNT,
-      );
-      setStakingDualInfos(
-        currentDualStakingInfos.concat(stakingDualInfosToAdd),
-      );
+  const stakingInfos = useMemo(() => {
+    return sortedLPStakingInfos
+      ? sortedLPStakingInfos.slice(
+          0,
+          pageIndex === 0 ? LOADFARM_COUNT : LOADFARM_COUNT * pageIndex,
+        )
+      : null;
+  }, [sortedLPStakingInfos, pageIndex]);
+
+  const stakingDualInfos = useMemo(() => {
+    return sortedStakingDualInfos
+      ? sortedStakingDualInfos.slice(
+          0,
+          pageIndex === 0 ? LOADFARM_COUNT : LOADFARM_COUNT * pageIndex,
+        )
+      : null;
+  }, [sortedStakingDualInfos, pageIndex]);
+
+  const getPoolApy = (pairId: string) => {
+    if (!pairId || !bulkPairs) {
+      return 0;
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [pageIndex]);
 
-  const stakingAPYs = useMemo(() => {
-    const sortedStakingInfos =
-      farmIndex === GlobalConst.farmIndex.LPFARM_INDEX
-        ? stakingInfos
-        : stakingDualInfos;
-    if (bulkPairs && sortedStakingInfos && sortedStakingInfos.length > 0) {
-      return sortedStakingInfos.map((info: any) => {
-        const oneDayVolume = bulkPairs[info.pair]?.oneDayVolumeUSD;
-        const reserveUSD = bulkPairs[info.pair]?.reserveUSD;
-        if (oneDayVolume && reserveUSD) {
-          const oneYearFeeAPY = getOneYearFee(oneDayVolume, reserveUSD);
-          return oneYearFeeAPY;
-        } else {
-          return 0;
-        }
-      });
+    const oneDayVolume = bulkPairs?.[pairId]?.oneDayVolumeUSD;
+    const reserveUSD = bulkPairs?.[pairId]?.reserveUSD;
+    if (oneDayVolume && reserveUSD) {
```

Comment:
reduce nesting   

`if(!oneDayVolume || !reserveUSD) { return 0; }`

Reply:
removed if else nesting by moving checks inside the functions: `getOneYearFee(oneDayVolume, reserveUSD)` itself .